include/deal.II/hp/fe_collection.h

```diff
@@ -371,6 +413,60 @@ namespace hp
     find_dominating_fe_in_subset(const std::set<unsigned int> &fes,
                                  const unsigned int            codim = 0) const;
 
+    /**
+     * Set functions determining the hierarchy of finite elements, i.e. a
+     * function @p next that returns the index of the finite element following
+     * the given one, and a function @p prev returning the preceding one.
+     *
+     * Both functions expect an hp::FECollection to be passed along with a
+     * finite element index, on whose basis the new index will be found and
+     * returned.
+     *
+     * @note Both passed and returned indices have to be valid within the index
+     * range of this collection, i.e. within [0, size()).
+     */
+    void
+    set_hierarchy(const std::function<unsigned int(
+                    const typename hp::FECollection<dim, spacedim> &,
+                    const unsigned int)> &next,
+                  const std::function<unsigned int(
+                    const typename hp::FECollection<dim, spacedim> &,
+                    const unsigned int)> &prev);
+
+    /**
+     * Set the default hierarchy corresponding to the index of each finite
+     * element in the collection.
+     *
+     * This default hierarchy is established with functions
+     * Hierarchy::next_index() and Hierarchy::previous_index().
+     */
+    void
+    default_hierarchy();
```

Comment:
call it `set_default_hierarchy()` to keep the symmetry with the previous function.